scripts/print-tags.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python3
+
+import sys
+
+TA_NAME, TA_LINE, TA_LOCAL, TA_POS, TA_TYPE, TA_ARGLIST, TA_SCOPE, \
+	TA_VARTYPE, TA_INHERITS, TA_TIME, TA_ACCESS, TA_IMPL, TA_LANG, \
+	TA_INACTIVE, TA_POINTER = range(200, 215)
+
+types = [
+	"undef", "class", "enum", "enumerator", "field", "function",
+	"interface", "member", "method", "namespace", "package",
+	"prototype", "struct", "typedef", "union", "variable", "externvar",
+	"macro", "macro_arg", "file", "other",
+	"UNKNOWN", "UNKNOWN", "UNKNOWN", "UNKNOWN", "UNKNOWN", "UNKNOWN",
+]
+
+type_dct = {0: types[0]}
+for i in range(len(types) - 1):
+	type_dct[1 << i] = types[i + 1]
+
+
+def decode_kind(kind, value):
+	val = ''
+	if kind == TA_NAME:
+		val = value.decode('utf-8')
+	elif kind == TA_LINE:
+		val = int(value)
+	elif kind == TA_LOCAL:
+		val = int(value)
+	elif kind == TA_TYPE:
+		val = type_dct[int(value)]
+	elif kind == TA_ARGLIST:
+		val = value.decode('utf-8')
+	elif kind == TA_SCOPE:
+		val = value.decode('utf-8')
+	elif kind == TA_POINTER:
+		val = int(value)
+	elif kind == TA_VARTYPE:
+		val = value.decode('utf-8')
+	elif kind == TA_INHERITS:
+		val = value.decode('utf-8')
+	elif kind == TA_TIME:
+		pass;
+	elif kind == TA_LANG:
+		pass;
+	elif kind == TA_INACTIVE:
+		pass;
+	elif kind == TA_ACCESS:
+		val = value.decode('utf-8')
+	elif kind == TA_IMPL:
+		val = value.decode('utf-8')
+	return val
+
+
+def print_tag(tag):
+	res = '{:<12}'.format(tag[TA_TYPE] + ': ')
+	if TA_VARTYPE in tag:
+		res += tag[TA_VARTYPE] + ' '
+	if TA_SCOPE in tag:
+		res += tag[TA_SCOPE] + ' :: '
+	res += tag[TA_NAME]
+	if TA_ARGLIST in tag:
+		res += tag[TA_ARGLIST]
+	if TA_INHERITS in tag:
+		res += ' extends ' + tag[TA_INHERITS]
+	if TA_POINTER in tag and tag[TA_POINTER] > 0:
+		res += '    flags: ' + str(tag[TA_POINTER])
+	sys.stdout.buffer.write(res.encode('utf-8'))
```

Comment:
I assume we are using `stdout.buffer` here because we are writing binary data to `stdout` and docs suggest to use `stdout.buffer` in such cases?
If so, maybe a comment could document this so we won't ask ourselves again in a year or so :).

Reply:
> I assume we are using stdout.buffer here because we are writing binary data to stdout and docs suggest to use stdout.buffer in such cases?

To be precise, after a failed attempt to write to stdout normally and trying to figure out what I was doing wrong, yes :-).

> If so, maybe a comment could document this so we won't ask ourselves again in a year or so :).

Will do.

Reply:
Done.